docs/config.md

```diff
@@ -3,89 +3,68 @@
+Once you have a network that you want to host the cluster on, you have to configure it as `externalNetworkId` in `cluster.yaml`.
 
 ## Public Network
 
-If your openstack cluster does not already have a public network, you should contact your cloud service provider. We will not review how to troubeshoot this here.
-
-## Floating IP
+If your openstack cluster does not already have a public network, you should contact your cloud service provider. We will not review how to troubleshoot this here.
 
-Create a Floating IP via OpenStack for the master node before creating a cluster with the clusterctl tool. This IP will be used by ``clusterctl`` to access the master node via SSH to retrieve the kubeconfig file generated by kubeadm on the master node.
+## Floating IPs
 
-   For example, ``openstack floating ip create <public_net>`` to create a floating ip, e.g:
-
-   ```bash
-   +--------------------------------------+------------------+---------------------+---------+
-   | id                                   | fixed_ip_address | floating_ip_address | port_id |
-   +--------------------------------------+------------------+---------------------+---------+
-   | aeefa79a-0e76-4c14-9da7-e5a9a6cc3787 |                  | 172.17.0.117        |         |
-   +--------------------------------------+------------------+---------------------+---------+
-   ```
+You have to be able to at least assign floating IPs in your OpenStack. If you don't have rights 
+to create floating IPs you have to make sure they already exist before creating the cluster.
 
-Once you have an available floating ip, then you can add it to the `machines.yaml` script where it says `<Available Floating IP>`. You only need to create and use one floating ip.
-
-## Proper Routing
-
-Your kubernetes cluster must be reachable from wherever cluster-api-provider-openstack is being run from to set it up, and probably needs to be reachable by external trafic for use. To make your cluster reachable by external traffic, you will need to set up an openstack router that connects your private network to your public network. For this example, lets say you have a subnet named ``kube-nodes-subnet`` in the private network you created, and a public network named ``public`` that you are trying to connect with a router named ``kube-router``.
-
-```bash
-openstack router create kube-router
-openstack router set kube-router --external-gateway public
-openstack router add subnet kube-router kube-nodes-subnet
-```
-
-For another example on networking in openstack, look here https://developer.openstack.org/firstapp-libcloud/networking.html
+There are different places where the floating IP has to be configured:
+* single-node control plane:
+  * Add a `.spec.floatingIP` field to the `<cluster-name>-controlplane-0` Machine in`controlplane.yaml`.
```

Comment:
There is a field `controlPlaneEndpoint: <floating ip>:6443` which I suppose is mandatory to be edited. I am a little confused where exactly do we need to enter floatingIP spec.

And wouldn't it be a nicer idea to add that in the controlplane template

Reply:
because the info is needed to connect to the controller IP, we have floating ip but that's in master machine ,so from cluster controller (actuator) you don't have the info unless you use k8s API to query all machines

Reply:
I added that this field must also be configured. Yeah the problem is that the kubeadm bootstrapper also needs this information and it doesn't watches our Machine resources (Also it's really OpenStack specific to have a master node with a floating IP)

Reply:
@jichenjc @sbueringer Got it Thanks.